components/board/collector.go

```diff
@@ -26,78 +27,66 @@ func (m method) String() string {
-		var readings []AnalogRecord
-		for k := range arg {
-			if reader, ok := board.AnalogReaderByName(k); ok {
-				value, err := reader.Read(ctx, data.FromDMExtraMap)
-				if err != nil {
-					// A modular filter component can be created to filter the readings from a component. The error ErrNoCaptureToStore
-					// is used in the datamanager to exclude readings from being captured and stored.
-					if errors.Is(err, data.ErrNoCaptureToStore) {
-						return nil, err
-					}
-					return nil, data.FailedToReadErr(params.ComponentName, analogs.String(), err)
+		var value int
+		if _, ok := arg["reader_name"]; !ok {
+			return nil, data.FailedToReadErr(params.ComponentName, analogs.String(),
+				errors.New("Must supply reader_name for analog collector"))
+		}
+		if reader, ok := board.AnalogReaderByName(arg["reader_name"].String()); ok {
+			value, err = reader.Read(ctx, data.FromDMExtraMap)
+			if err != nil {
+				// A modular filter component can be created to filter the readings from a component. The error ErrNoCaptureToStore
+				// is used in the datamanager to exclude readings from being captured and stored.
+				if errors.Is(err, data.ErrNoCaptureToStore) {
+					return nil, err
 				}
-				readings = append(readings, AnalogRecord{AnalogName: k, AnalogValue: value})
+				return nil, data.FailedToReadErr(params.ComponentName, analogs.String(), err)
 			}
 		}
-		return AnalogRecords{Readings: readings}, nil
+		return pb.ReadAnalogReaderResponse{
+			Value: int32(value),
+		}, nil
 	})
 	return data.NewCollector(cFunc, params)
 }
 
-func newGPIOCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
+// NewGPIOCollector returns a collector to register a gpio get method. If one is already registered
+// with the same MethodMetadata it will panic.
+func NewGPIOCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
 	board, err := assertBoard(resource)
 	if err != nil {
 		return nil, err
 	}
 
 	cFunc := data.CaptureFunc(func(ctx context.Context, arg map[string]*anypb.Any) (interface{}, error) {
-		var readings []GpioRecord
-		for k := range arg {
-			if gpio, err := board.GPIOPinByName(k); err == nil {
-				value, err := gpio.Get(ctx, data.FromDMExtraMap)
-				if err != nil {
-					// A modular filter component can be created to filter the readings from a component. The error ErrNoCaptureToStore
-					// is used in the datamanager to exclude readings from being captured and stored.
-					if errors.Is(err, data.ErrNoCaptureToStore) {
-						return nil, err
-					}
-					return nil, data.FailedToReadErr(params.ComponentName, gpios.String(), err)
+		var value bool
+		if _, ok := arg["reader_name"]; !ok {
+			return nil, data.FailedToReadErr(params.ComponentName, gpios.String(),
+				errors.New("Must supply reader_name for gpio collector"))
+		}
+		if gpio, err := board.GPIOPinByName(arg["reader_name"].String()); err == nil {
```

Comment:
I think it makes sense for the above AnalogCollector to be `reader_name` but here to be `pin_name`.